runway/cfngin/hooks/aws_lambda.py

```diff
@@ -322,10 +615,19 @@ def _upload_function(s3_conn, bucket, prefix, name, options, follow_symlinks,
     # absolute path, which is exactly what we want.
     if not os.path.isabs(root):
         root = os.path.abspath(os.path.join(get_config_directory(), root))
-    zip_contents, content_hash = _zip_from_file_patterns(root,
-                                                         includes,
-                                                         excludes,
-                                                         follow_symlinks)
+    if options.get('dockerize_pip'):
```

Comment:
@ITProKyle we had discussed leaving this as is, and then changing in the near future when we need to be able to use pip without docker, but I'm still having some reservation about making the change after release.

I'm not wholly against having a different default than the sls plugin, but I'm still not quite convinced that it's worth having it be [different](https://github.com/UnitedIncome/serverless-python-requirements#install). I'd rather we align it (e.g. check here for a Pipfile.lock or requirements.txt, force users to set the docker option if they want to use it, etc) for consistency.

Reply:
ok, so you are thinking:

- always try to use pip - don't require docker
  - check for/use `requirements.txt` unless told to use something else
  - check for/use `Pipfile` if the above is not found or explicitly told to do so
- leave docker as is, just don't lock the use of pip behind it

If we're going this route, it may also be worth adding a `python_path` argument to handle `python`/`python3`/etc path differences [sls does similar](https://github.com/UnitedIncome/serverless-python-requirements#customize-python-executable). we can default to using `sys.executable` if this is not provided.

We would also need to include handling for our `pyinstaller` builds. do we disable this if using `pyinstaller`? require `python_path`? include pip and pipenv in the build?

Reply:
Agreed on all accounts.

> include pip and pipenv in the build so we can use run-python?
I'll admit the idea crossed my mind but I wasn't sure if I'd be asking for too much :)

Reply:
> include pip and pipenv in the build so we can use run-python

Since this will _probably_ be an edge case, think it can slide to a later release so we can get this one out? The other changes will are doable in a day since most of the logic is already there, just needs some rearranging.

Reply:
Yeah, sounds good, I agree. I think it'll be a really nice featureset to have but it's seems additive (i.e. won't change anyone's use/configs after it's added & will just enable more uses of it) to me and fine to add in the next release